src/lib/FixedLatencyMemoryInterface.cc

```diff
@@ -20,34 +23,32 @@ void FixedLatencyMemoryInterface::tick() {
     }
 
     const auto& target = request.target;
+    uint64_t requestId = request.requestId;
 
     if (request.write) {
-      // Write: write data directly to memory
-      assert(target.address + target.size <= memory_->getMemorySize() &&
-             "Attempted to write beyond memory limit");
-
-      simeng::memory::WriteRespPacket* resp =
-          (simeng::memory::WriteRespPacket*)memory_->requestAccess(
-              new simeng::memory::WritePacket{
-                  target.address, target.size,
-                  request.data.getAsVector<char>()});
-      delete resp;
+      auto fn = [&](memory::DataPacket* dpkt = NULL) -> void {
```

Comment:
This was done just so that the integration with SST doesn't require any major refactoring, this way the callback function can be overriden to work with the SST memory. It was done to make SST less painful. I don't see any other way of doing it, other than implementing a custom event system or having an internal buffer inside MMU (which would add another O(n) memory, bearing in mind these requests are already stored in multiple buffers inside of SST). Since we want to refactor memory interface anyway in future iteration this was the simplest way to achieve functionality and reducing refactoring overhead.